src/Compilers/CSharp/Portable/Binder/ForEachLoopBinder.cs

```diff
@@ -855,7 +855,7 @@ private void GetDisposalInfoForEnumerator(ref ForEachEnumeratorInfo.Builder buil
             // For async foreach, we don't do the runtime check
             if ((!enumeratorType.IsSealed && !isAsync) ||
                 this.Conversions.ClassifyImplicitConversionFromType(enumeratorType,
-                    isAsync ? this.Compilation.GetWellKnownType(WellKnownType.System_IAsyncDisposable) : this.Compilation.GetSpecialType(SpecialType.System_IDisposable),
+                    isAsync ? this.Compilation.GetWellKnownType(WellKnownType.System_IAsyncDisposable, recordUsage: false) : this.Compilation.GetSpecialType(SpecialType.System_IDisposable),
```

Comment:
>System_IAsyncDisposable [](start = 78, length = 23)

I assume we don't need to record this because we're also using IAsyncEnumerable or IAsyncEnumerator, which depend on this type, so will record usage?

Reply:
> I assume we don't need to record this because we're also using IAsyncEnumerable or IAsyncEnumerator, which depend on this type, so will record usage?

An attempt to classify conversion is not usage. Usage would be binding a conversion. In addition, we are using ```enumeratorType``` and for that type all bases (including implemented interfaces) are considered used. 

---
In reply to: [350910870](https://github.com/dotnet/roslyn/pull/40010#discussion_r350910870) [](ancestors = 350910870)